packages/teuchos/numerics/src/Teuchos_SerialDenseMatrix.hpp

```diff
@@ -364,10 +364,14 @@ class SerialDenseMatrix : public CompObject, public BLAS<OrdinalType, ScalarType
                  const OrdinalType numCols)
   {
     const size_t size = size_t(numRows) * size_t(numCols);
+#ifndef _WIN32
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wvla"
+#endif
     return new ScalarType[size];
+#ifndef _WIN32
 #pragma GCC diagnostic pop
+#endif
```

Comment:
The statement `new ScalarType[size]` is not a variable length array (i.e. that is allocated on the stack).  This is most definingly allocated on the heap and this is perfectly acceptable standard C++ as far as I know.  Therefore, these `#pragma` commands for:

> `-Wvla`
> Warn if variable length array is used in the code. `-Wno-vla` will prevent the `-pedantic` warning of the variable length array.

should not be needed (see [here](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html) and [here](https://gcc.gnu.org/onlinedocs/gcc/Variable-Length.html#Variable-Length)).

Can we just strip all of these `#pragma` statements out?

Reply:
Gladly!

Reply:
A long time ago, this did actually produce warnings with nvcc: https://github.com/trilinos/Trilinos/pull/6494#discussion_r360974850 (found that from the blame). At some point I think this just got fixed in nvcc because it's not happening for me now with cuda 12.

Reply:
NVCC and not understanding C++ 👎(It is so frustrating when you are trying to get rid of warnings with NVCC and then you realize the code is 100% standards compliant and there is absolutely nothing wrong with it and no other compiler in existence would produce a warning for that.)